### PR TITLE
fill product with simulation installment

### PR DESCRIPTION
**Pull request by hiagolcm**
#### What problem is this solving?

The simulation API is not replacing the biggyproduct installment.

#### How should this be manually tested?

[Workspace](https://hiago--tfbsmy.myvtex.com/camiseta%20manga%20curta%20estampada?_q=CAMISETA%20MANGA%20CURTA%20ESTAMPADA&map=ft&page=1)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: node/typings/Checkout.ts [hunk 1]
```diff
@@ -60,6 +60,60 @@ interface OrderFormItem {
...
+    description: string | null
+    requiresAuthentication: boolean
+    dueDate: string
+    availablePayments: any | null
+  }[]
+  payments: any[]
+  giftCards: any[]
+  giftCardMessages: any[]
+  availableAccounts: any[]
+  availableTokens: any[]
```
Comment (thalytafabrine) [inline node/typings/Checkout.ts:114]: can you create types for these props?

File: node/commons/compatibility-layer.ts [hunk 2]
```diff
@@ -152,6 +152,30 @@ const fillSearchItemWithSimulation = (searchItem: SearchItem, orderFormItems: Or
...
+        if ((previous.hasInterestRate === current.hasInterestRate) && current.count > previous.count) {
+          return current
+        }
+
+        return previous
+      })
+
+      seller.commertialOffer.Installments = [{
+        Value: correctInstallment.value / 100,
+        InterestRate: 0,
```
Comment (LucasTornai) [inline node/commons/compatibility-layer.ts]: There will always be a installment without InterestRate? Else this field should change accordingly

File: node/typings/Checkout.ts (hunk 1 above)
Comment (hiagolcm) [reply node/typings/Checkout.ts:114]: The API is not returning these values, so I can't complete the types 😢

File: node/commons/compatibility-layer.ts (hunk 2 above)
Comment (hiagolcm) [reply node/commons/compatibility-layer.ts]: Done

File: node/commons/compatibility-layer.ts [hunk 3]
```diff
@@ -152,6 +152,30 @@ const fillSearchItemWithSimulation = (searchItem: SearchItem, orderFormItems: Or
...
+      })
+
+      seller.commertialOffer.Installments = [{
+        Value: correctInstallment.value / 100,
+        InterestRate: correctInstallment.interestRate,
+        TotalValuePlusInterestRate: correctInstallment.total / 100,
+        NumberOfInstallments: correctInstallment.count,
+        Name: '',
+        PaymentSystemName: '',
+        PaymentSystemGroupName: '',
```
Comment (klzns) [inline node/commons/compatibility-layer.ts:177]: Why those fields are not filled?

**Review by thalytafabrine — APPROVED** [review]

**Review by LucasTornai — APPROVED** [review]

**Pull request merged**